packages/x-date-pickers/src/internals/hooks/useField/useFieldCharacterEditing.ts

```diff
@@ -149,71 +151,112 @@ export const useFieldCharacterEditing = <TDate, TSection extends FieldSection>({
     return queryResponse;
   };
 
-  const applyLetterEditing: CharacterEditingApplier<TDate, TSection> = (params) => {
+  const applyLetterEditing: CharacterEditingApplier<TDate> = (params) => {
+    const findMatchingOptions = (
+      format: string,
+      options: string[],
+      queryValue: string,
+    ): ReturnType<QueryApplier<TSection>> => {
+      const matchingValues = options.filter((month) => month.toLowerCase().startsWith(queryValue));
```

Comment:
Or am I missing something and `month` makes sense here? 🤔 
```suggestion
      const matchingValues = options.filter((option) => option.toLowerCase().startsWith(queryValue));
```